### PR TITLE
docs/config-examples: generate dynamically from a toml list

### PR DESCRIPTION
**Pull request by MattSturgeon**
This should help to maintain uniformity and hopefully make contributions easier.

I would like to add tests for robustness and to avoid regressions. Any ideas on what to test and/or how would be great.

If there's any toml validation that needs to happen, maybe we could have a test for that too? This could then be included in the pre-commit hooks.

EDIT: I added some regression tests that ensure it is able to handle a few edge-cases.

### CODE REVIEW

File: user-configs/list.toml
Comment (MattSturgeon) [inline user-configs/list.toml:1]: `/user-configs/list.toml` is meant to be easy for users to discover, but maybe putting it in `/docs` is neater?

Comment (GaetanLepage) [reply user-configs/list.toml:1]: Yes, I think it makes more sense in `/docs` and would pollute less the root of the tree.

**Review by GaetanLepage — DISMISSED** [review]
Very good idea Matt !

**Comment by MattSturgeon** [comment]
> Does nixpkgs have a two derivations and/or files should produce the same output test helper?

Looks like the answer is [`testEqualContents`](https://nixos.org/manual/nixpkgs/unstable/#tester-testEqualContents)

File: docs/mdbook/default.nix [hunk 1]
```diff
@@ -398,5 +403,6 @@ pkgs.stdenv.mkDerivation (finalAttrs: {
             echo "- The $link, for use with nixpkgs \`$nixpkgs\`$suffix" >> "$out"
           done
         '';
+    user-configs = callPackage ../user-configs { };
```
Comment (MattSturgeon) [inline docs/mdbook/default.nix:406]: Hm... Not sure how I feel about naming here.

The page in the docs is "config-examples.md" while the package that's generating them is "user-configs".

Comment (GaetanLepage) [reply docs/mdbook/default.nix:406]: No hard feeling, both are fine.

**Review by khaneliman — APPROVED** [review]
Love it, thanks :)

File: docs/mdbook/default.nix (hunk 1 above)
Comment (MattSturgeon) [reply docs/mdbook/default.nix:406]: Ok, I'll leave it as-is then. Having a different name for the actual page in the docs and the packaged used to construct the main part of that page isn't the end of the world.

**Comment by MattSturgeon** [comment]
@mergify queue
